src/Util/MIMETypeUtil.m

```diff
@@ -294,9 +289,18 @@ + (NSString *)filePathForAttachment:(NSString *)uniqueId
     } else if ([self isBinaryData:contentType]) {
         return [MIMETypeUtil filePathForBinaryData:uniqueId ofMIMEType:contentType inFolder:folder];
     } else if ([contentType isEqualToString:OWSMimeTypeOversizeTextMessage]) {
-        return [MIMETypeUtil filePathForOversizeTextMessage:uniqueId inFolder:folder];
+        // This file extension is arbitrary - it should never be exposed to the user or
+        // be used outside the app.
+        return [self filePathForData:uniqueId withFileExtension:@"signal-text-message" inFolder:folder];
     } else if ([contentType isEqualToString:OWSMimeTypeUnknownForTests]) {
-        return [MIMETypeUtil filePathForUnknownContent:uniqueId inFolder:folder];
+        // This file extension is arbitrary - it should never be exposed to the user or
+        // be used outside the app.
+        return [self filePathForData:uniqueId withFileExtension:@"unknown" inFolder:folder];
+    }
+
+    NSString *fileExtension = [self fileExtensionForMIMEType:contentType];
+    if (fileExtension) {
+        return [self filePathForData:uniqueId withFileExtension:fileExtension inFolder:folder];
```

Comment:
Eventually, we'll want to honor the file extension (and probably the name as well) of the sent attachment, but for now we need to make sure that the attachment's data is saved to disk with an appropriate file extension, since the file extension is what `UIActivityViewController` uses to determine which kinds of sharing are appropriate.